internal/cmd/gocdk/serve.go

```diff
@@ -91,47 +128,73 @@ func serveLoop(ctx context.Context, pctx *processContext, logger *log.Logger, ad
 	}()
 
 	// Apply Terraform configuration in biome.
-	if err := apply(ctx, pctx, []string{biome}); err != nil {
+	if err := apply(ctx, pctx, []string{opts.biome}); err != nil {
 		return err
 	}
 
 	// Build and run the server.
-	// TODO(light): Bind host is ignored for now because eventually this
-	// subprocess will be reverse proxied. The proxy should respect the host.
-	_, portString, err := net.SplitHostPort(address)
-	if err != nil {
-		return err
-	}
-	port, err := net.LookupPort("tcp", portString)
-	if err != nil {
-		return xerrors.Errorf("resolve port for %q: %w", address, err)
-	}
-	alloc := &serverAlloc{
-		exePath: filepath.Join(buildDir, "server"),
-		port:    port,
+	allocA := &serverAlloc{
+		exePath: filepath.Join(buildDir, "serverA"),
+		port:    opts.basePort,
 	}
-	logger.Println("Building server...")
-	if err := buildForServe(ctx, pctx, moduleRoot, alloc.exePath); err != nil {
-		return err
+	allocB := &serverAlloc{
+		exePath: filepath.Join(buildDir, "serverB"),
+		port:    opts.basePort + 1,
 	}
-	process, err := alloc.start(ctx, pctx, logger)
-	if err != nil {
-		return err
+	currAlloc, nextAlloc := allocA, allocB
```

Comment:
Nit: I found the name of these variables to be backwards (you build/run a new server into `curAlloc`, while `nextAlloc` has the currently running server most of the time).

Reply:
Ack, thanks for pointing that out. What about `buildAlloc` and `otherAlloc`?

Reply:
How about `liveAlloc` and `spareAlloc`? (where you build into `spareAlloc`, and then swap it to `liveAlloc` once it's up).

Reply:
I like that naming, but I'm concerned about a slight imprecision. At the start, `liveAlloc` will not represent a live server, but in steady state it behaves as you say.

I'd be fine using the naming if you and @clausti are okay with it. It does seem strictly better than `currAlloc` and `nextAlloc`.

Reply:
I agree that `liveAlloc` and `spareAlloc` are easier to make sense of than `current` and `next`

Reply:
Done.